populus/compilation/backends/base.py

```diff
@@ -4,12 +4,45 @@
     get_import_path,
 )
 
+from semantic_version import (
+    Spec,
+    Version,
+)
+
+from solc import (
+    get_solc_version,
+)
+
+
+
+#TODO-PRIORITY-01:
+# move to new global compilation settings
+# by solc version
+
+DEFAULT_SOLC_SETTINGS = {
+    'optimize': True,
+    'output_values': ['abi', 'bin', 'bin-runtime', 'metadata']
+}
+
 
 class BaseCompilerBackend(object):
     compiler_settings = None
+    solc_version = None
+
+    def __init__(self, settings=None):
+
+        try:
+            assert get_solc_version() in Spec(self.solc_version)
+        except AssertionError:
+            raise OSError(
+                "The 'SolcStandardJSONBackend can only be used with solc versions {version}.".format(
+                    version=self.solc_version
+                )
+            )
+        except:
+            raise
 
-    def __init__(self, settings):
-        self.compiler_settings = settings
+        self.compiler_settings = DEFAULT_SOLC_SETTINGS if settings == None else settings
```

Comment:
I'm not a big fan of inline turnary statements.  They are not great for readability.  Maybe expand this to it's 4-line equivalent?